docpages/example_programs/interactions_and_components/private-messaging.md

```diff
@@ -0,0 +1,80 @@
+\page private-messaging Sending private messages
+
+Sometimes it's simply not enough to ping someone in a server with a message, and we get that. That's why you can private message people! This tutorial will cover how to make a command that will either message the author of the command or message a specified user! This is done with a command option! So, let's get into it!
+
+\note This tutorial makes use of callbacks. For more information about that, visit \ref callback-functions "Using Callback Functions".
+
+~~~~~~~~~~{.cpp}
+#include <dpp/dpp.h>
+
+int main()
+{
+    /* Create the bot */
+    dpp::cluster bot("token");
+
+    bot.on_log(dpp::utility::cout_logger());
+
+    /* The event is fired when someone issues your commands */
+    bot.on_slashcommand([&bot](const dpp::slashcommand_t & event) {
+
+        /* Check which command they ran */
+        if (event.command.get_command_name() == "pm") {
+
+            /* If there was no input, we want to only send a message to the command author. */
+            if(event.get_parameter("user").index() == 0) {
+
+                /* Send a message to the command author. */
+                bot.direct_message_create(event.command.get_issuing_user().id, dpp::message("Here's a private message!"), [event](const dpp::confirmation_callback_t& callback) {
+                    if(callback.is_error()) {
+                        event.reply(dpp::message("I couldn't send a message to you.").set_flags(dpp::m_ephemeral));
+                        return;
+                    }
+
+                    event.reply(dpp::message("I've sent you a private message.").set_flags(dpp::m_ephemeral));
+                });
+
+                /* Because we've sent the message, let's stop the function now. */
+                return;
+            }
+
+            /* Here, we can assume that there was a user inputted, so we can just retrieve that data! */
+
+            /* Get the user specified by the command author. */
+            auto user = std::get<dpp::snowflake>(event.get_parameter("user"));
```

Comment:
instead of repeating the code block twice, consider just setting user to the current user if the parameter is not specified?

Reply:
This was considered but I thought it was more beginner-friendly to do it this way. Can be rectified if you would like :)

Reply:
To further explain, I thought it was better to be explicit to beginners about only messaging the command author if there was nothing inputted, also meaning they can understand that we do a whole different thing if no user was specified, as opposed to sending a message to a specified user.

Whilst we do expect people to have an understanding of C++, we encounter many people who don't so I want to still cater to those, slightly, by making this extremely idiomatic. If this was personal code, I would have done as suggested :)

Reply:
yeah it's better to teach newbies to DRY

Reply:
Okie dokie! Will rectify this then.